packages/api-client/src/views/Request/hooks/useOpenApiWatcher.ts

```diff
@@ -0,0 +1,181 @@
+import { useWorkspace } from '@/store'
+import { specDictionary } from '@/store/import-spec'
+import {
+  combineRenameDiffs,
+  diffToCollectionPayload,
+  diffToRequestPayload,
+  diffToSecuritySchemePayload,
+  diffToServerPayload,
+  diffToTagPayload,
+} from '@/views/Request/libs/live-sync'
+import { createExampleFromRequest } from '@scalar/oas-utils/entities/spec'
+import { createHash, fetchSpecFromUrl } from '@scalar/oas-utils/helpers'
+import { parseSchema } from '@scalar/oas-utils/transforms'
+import { useTimeoutPoll } from '@vueuse/core'
+import microdiff, { type Difference } from 'microdiff'
+import { watch } from 'vue'
+
+/** Live Sync polling timeout */
+const FIVE_SECONDS = 5 * 1000
+
+/**
+ * Hook which handles polling the documentUrl for changes then attempts to merge what is new
+ *
+ * TODO:
+ * - check lastModified or similar headers
+ * - speed up the polling when there's a change then slowly slow it down
+ */
+export const useOpenApiWatcher = () => {
+  const {
+    activeCollection,
+    activeWorkspace,
+    collectionMutators,
+    requests,
+    requestMutators,
+    requestExamples,
+    requestExampleMutators,
+    securitySchemes,
+    securitySchemeMutators,
+    servers,
+    serverMutators,
+    tags,
+    tagMutators,
+  } = useWorkspace()
+
+  // Transforms and applies the diff to our mutators
+  const applyDiff = (d: Difference) => {
+    if (!d.path.length || !activeCollection.value?.uid) return
+
+    // Info/Security
+    if (d.path[0] === 'info' || d.path[0] === 'security') {
+      const payload = diffToCollectionPayload(d, activeCollection.value)
+      if (payload) collectionMutators.edit(...payload)
+    }
+    // Components.securitySchemes
+    else if (d.path[0] === 'components' && d.path[1] === 'securitySchemes') {
+      const payload = diffToSecuritySchemePayload(
+        d,
+        activeCollection.value,
+        securitySchemes,
+      )
+
+      if (payload?.method === 'add') securitySchemeMutators.add(...payload.args)
```

Comment:
typescript did not do what I wanted it to here unfortunately so it gets pretty ugly. We can fix it up later I think